core/core.go

```diff
@@ -307,12 +306,22 @@ func New(config Config) (*App, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	// Remove any old mini headers that might be lingering in the database.
+	// HACK(albrow): This is a workaround for an issue where miniheaders are
```

Comment:
We should update this comment. I realized we should actually leave this in place for longer (maybe indefinitely) since

1. It is still necessary for anyone upgrading from older versions to >= 9.0.1 in the future.
2. There's still a chance there are old MiniHeaders in the database (e.g. due to a sudden unexpected shut down).